Doc/library/annotationlib.rst

```diff
@@ -0,0 +1,178 @@
+
+   .. versionadded:: 3.14
+
+.. class:: ForwardRef
+
+   A proxy object for forward references in annotations.
+
+   Instances of this class are returned when the :attr:`~Format.FORWARDREF` format is
+   used and annotations contain a name that cannot be resolved.  This can happen
+   when a forward reference is used in an annotation, such as when a class is
+   referenced before it is defined.
+
+   .. attribute:: __forward_arg__
+
+      A string containing the code that was evaluated to produce the :class:`~ForwardRef`.
+      The string may not be exactly equivalent to the original source.
+
+   .. method:: evaluate(*, globals=None, locals=None, type_params=None, owner=None)
+
+      Evaluate the forward reference, returning its value.
+
+      This may throw an exception such as :exc:`NameError` if the forward reference
+      refers to names that do not exist. The parameters to the function can be used to
+      provide bindings for names that would otherwise be undefined.
+
+      *globals* and *locals* are passed to :func:`eval()`, representing the global and
+      local namespaces in which the name is evaluated. *type_params*, if given, must be
+      a tuple of :ref:`type parameters <type-params>` that are in scope while the forward
+      reference is being evaluated. *owner* is the object that owns the annotation from
+      which the forward reference derives, usually a function, class, or module.
+      :class:`~ForwardRef` instances returned by :func:`get_annotations` retain
+      a reference to their owner, so it is not necessary to pass it in explicitly.
+
+      Once a :class:`~ForwardRef` instance has been evaluated, it caches the evaluated
+      value, and future calls to :meth:`evaluate` will return the cached value, regardless
+      of the parameters passed in.
+
+   .. versionadded:: 3.14
+
+.. function:: get_annotate_function(obj)
+
+   Retrieve the :term:`annotate function` for *obj*. Return ``None`` if *obj* does not have an
+   annotate function.
+
+   This is usually equivalent to accessing the :attr:`~object.__annotate__` attribute of *obj*,
+   but direct access to the attribute may return the wrong object in certain situations involving
+   metaclasses. It is recommended to use this function instead of accessing the attribute directly.
+
+   .. versionadded:: 3.14
+
+.. function:: get_annotations(obj, *, globals=None, locals=None, eval_str=False, format=Format.VALUE)
+
+   Compute the annotations dict for an object.
+
+   ``obj`` may be a callable, class, module, or other object with
+   :attr:`~object.__annotate__` and :attr:`~object.__annotations__` attributes.
+   Passing in an object of any other type raises :exc:`TypeError`.
+
+   The *format* parameter controls the format in which annotations are returned:
+
+   Returns a dict.  ``get_annotations()`` returns a new dict every time
+   it's called; calling it twice on the same object will return two
+   different but equivalent dicts.
```

Comment:
```suggestion
   .. note::

      This function returns a *new* dict every time it's called; calling it
      twice on the same object returns two different but equivalent dicts.
```

Reply:
Why should this be indented into a note? I feel that makes the text noisier.

Reply:
You'll have a nice box around this (important) behaviour. Since your bullet list contained "Always, always, always creates a new dict", I feel this behaviour should be highlighted. I used "note" but you can also use "important" or "tip".